File: src/types/workspace-file-tree.ts
export type WorkspaceFileTreeNodeKind = 'file' | 'dir';

export interface WorkspaceFileTreeNode {
  path: string;
  name: string;
  kind: WorkspaceFileTreeNodeKind;
  hasChildren: boolean;
  children?: WorkspaceFileTreeNode[];
}

export interface ListWorkspaceFileTreeInput {
  path?: string;
  depth?: number;
}

File: src-tauri/src/models/workspace_file_tree.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceFileTreeNode {
    pub path: String,
    pub name: String,
    pub kind: String,
    pub has_children: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub children: Option<Vec<WorkspaceFileTreeNode>>,
}
